tests/integration/test_search.py

```diff
@@ -48,3 +48,22 @@ def test_case_insensitive_doi(test_record):
 
     assert data_original['hits']
     assert data_original['hits'] == data_upper['hits']
+
+
+def test_source_field(app_client, test_record):
+    doi = get_value(test_record, 'dois[0].value')
+    search_param = '"%s"' % doi
+
+    full_response = app_client.get('/api/records/?q=%s' % quote(search_param))
+    full_data = json.loads(full_response.data)
+
+    partial_response = app_client.get('/api/records/?q=%s&_source=dois' % quote(search_param))
+    partial_data = json.loads(partial_response.data)
+
+    # drop everything that is not needed in full data
+    for hit in full_data['hits']['hits']:
+        for key in hit['metadata'].keys():
+            if key not in ('dois', 'control_number'):
+                hit['metadata'].pop(key)
+
+    assert full_data == partial_data
```

Comment:
This test is a bit weird, do you want to test that the search with and without the `_source` field are the same except for all the fields that are not `dois`?
An easier and cleaner way would be to just be sure that the search with `_source` returns something that contains the `dois` key and not another one, and of course checking that the DOI value is the one that you expect.